templates/s3/s3.py

```diff
@@ -377,6 +388,27 @@ def _deploy_minio_backend_namespaces(self):
 
         return deployed_namespaces
 
+
+    def _handle_data_shard_failure(self, connection_info):
+        namespace = self._get_namespace_by_address(connection_info['address'])
+
+        from time import sleep
```

Comment:
imports belongs at the top of the file

Reply:
check https://github.com/threefoldtech/0-robot/blob/development/docs/services/actions.md#retry-decorator 
that's what you're trying to implement here